lib/spaceship/client.rb

```diff
@@ -184,18 +196,36 @@ def login(user = nil, password = nil)
 
     def with_retry(tries = 5, &block)
       return block.call
-    rescue Faraday::Error::TimeoutError, AppleTimeoutError => ex # New Faraday version: Faraday::TimeoutError => ex
+    rescue Faraday::Error::ConnectionFailed, Faraday::Error::TimeoutError, AppleTimeoutError => ex # New Faraday version: Faraday::TimeoutError => ex
       unless (tries -= 1).zero?
         logger.warn("Timeout received: '#{ex.message}'.  Retrying after 3 seconds (remaining: #{tries})...")
         sleep 3
         retry
       end
-
+      raise ex # re-raise the exception
+    rescue UnauthorizedAccessError => ex
+      if @loggedin
+        msg = "Auth error received: '#{ex.message}'. Login in again then retrying after 3 seconds (remaining: #{tries})..."
+        puts msg
+        logger.warn msg
```

Comment:
Mostly for troubleshooting purposes while developing. The default spacesip log file is a bit verbose but something like this:
![screen shot 2016-02-06 at 14 29 25](https://cloud.githubusercontent.com/assets/24282/12875245/2fff3dde-cde6-11e5-88b0-db9ea7cf3f38.png)

makes it easy to follow what is going on. Although, the `fastlane_core` `Helper` isn't available because spaceship doesn't depend on `fastlane_core`.